controllers/awsmachine_controller.go

```diff
@@ -296,7 +305,11 @@ func (r *AWSMachineReconciler) reconcileDelete(machineScope *scope.MachineScope,
 	machineScope.Info("Handling deleted AWSMachine")
 
 	ec2Service := r.getEC2Service(ec2Scope)
-	secretSvc := r.getSecretsManagerService(clusterScope)
+	if clusterScope.SecretBackend() == "ssm" {
```

Comment:
Do you think it would be possible to have a single function call (i.e. something like `getSecretBackend`) that has the conditional logic to return the specific secrets backend?

Reply:
Would be in favour of this. Not least I suspect we'll need to add an S3 backend too for Ignition, which will solve things for @dongsupark